### PR TITLE
Fix PETScVector

### PR DESCRIPTION
**Pull request by chleh**
* fixes copy ctor and makes it explicit
* adds move ctor

### CODE REVIEW

**Comment by endJunction** [comment]
A note aside: is there an obvious reason not to implement the assignment operators?

File: MathLib/LinAlg/PETSc/PETScVector.cpp [hunk 1]
```diff
@@ -77,6 +77,17 @@ PETScVector::PETScVector(const PETScVector &existing_vec, const bool deep_copy)
     }
 }
 
+PETScVector::PETScVector(PETScVector &&other)
+    : _v{std::move(other._v)}
```
Comment (norihiro-w) [inline MathLib/LinAlg/PETSc/PETScVector.cpp]: Why are using {} instead of  ()? Make it consistent with another constructor

Comment (chleh) [inline MathLib/LinAlg/PETSc/PETScVector.cpp]: Isn't it consistent within this file?

Comment (norihiro-w) [inline MathLib/LinAlg/PETSc/PETScVector.cpp]: if you ask me so, my answer is "yes i do". Loosing consistency is not very minor thing for me.

Comment (chleh) [inline MathLib/LinAlg/PETSc/PETScVector.cpp]: Now all initializer lists in `PETScVector.*` use braces for construction.

**Comment by chleh** [comment]
@endJunction I deleted the move assignment operator explicitly as discussed.

**Comment by norihiro-w** [comment]
:+1:

**Pull request merged**